library/src/rocsparse_debug.cpp

```diff
@@ -84,6 +89,16 @@ void rocsparse::debug_variables_st::set_debug_verbose(bool value)
     }
 }
 
+void rocsparse::debug_variables_st::set_debug_warnings(bool value)
+{
+    if(value != this->debug_warnings)
+    {
+        s_mutex.lock();
```

Comment:
Why do we need a mutex?

Reply:
It was already there. 
The mutex is there for thread-safe setters.